src/character.cpp

```diff
@@ -596,6 +596,41 @@ std::vector<item_location> Character::nearby( const std::function<bool(const ite
     return res;
 }
 
+long int Character::i_add_to_container(const item &it, const bool unloading)
+{
+    long int charges = it.charges;
+    
+    if( !it.is_ammo() || unloading ) {
+        return charges;
+    }
+    
+    const itype_id item_type = it.typeId();
+    
+    auto add_to_container = [&it, &charges](item &container) {
+        if( container.contents.front().charges < container.ammo_capacity() ) {
```

Comment:
Would be clearer if you wrapped `container.contents.front()` in a reference.
Something like `contained_ammo` or something like that.